src/Microsoft.SqlTools.ManagedBatchParser/BatchParser/BatchParserWrapper.cs

```diff
@@ -109,7 +109,7 @@ private static List<BatchDefinition> ConvertToBatchDefinitionList(string content
                     if (count > 1)
                     {
 
-                        batchText = content.Substring(offsets[count-1], batchInfos[count - 1].length);
+                        batchText = batchInfos[count - 1].batchText;
                         BatchDefinition lastBatchDef = GetLastBatchDefinition(reader, batchInfos[count - 1], batchText);
```

Comment:
Are the changes in this file relevant to the sqlcmd stuff or just bugs that you're fixing in this PR?

Reply:
- BatchText is required for SQLCMD substituion of sqlcmd variables.
- +1 in start line is just to fix wrong start line in messages